datafusion/sqllogictest/test_files/string_view.slt

```diff
@@ -15,99 +15,232 @@
+
+
+########
+## StringView to Dictionary
+########
+
+# test StringViewArray with Dictionary columns
+query TTBBBB
+select
+  column1_utf8, column2_utf8,
+  column1_utf8view  = column2_dict,
+  column2_dict      = column1_utf8view,
+  column1_utf8view <> column2_dict,
+  column2_dict     <> column1_utf8view
+from test;
+----
+Andrew X false false true true
+Xiangpeng Xiangpeng true true false false
+Raphael R false false true true
+NULL R NULL NULL NULL NULL
+
+# StringView column to Dict scalar
+query TTBBBB
+select
+  column1_utf8, column2_utf8,
+  column1_utf8view                 = arrow_cast('Andrew', 'Dictionary(Int32, Utf8)'),
+  arrow_cast('Andrew', 'Dictionary(Int32, Utf8)')     = column1_utf8view,
+  column1_utf8view                 <> arrow_cast('Andrew', 'Dictionary(Int32, Utf8)'),
+  arrow_cast('Andrew', 'Dictionary(Int32, Utf8)')     <> column1_utf8view
+from test;
+----
+Andrew X true true false false
+Xiangpeng Xiangpeng false false true true
+Raphael R false false true true
+NULL R NULL NULL NULL NULL
+
+# Dict column to StringView scalar
+query TTBBBB
+select
+  column1_utf8, column2_utf8,
+  column1_dict                     = arrow_cast('Andrew', 'Utf8View'),
+  arrow_cast('Andrew', 'Utf8View') = column1_dict,
+  column1_dict                    <> arrow_cast('Andrew', 'Utf8View'),
+  arrow_cast('Andrew', 'Utf8View') <> column1_dict
+from test;
+----
+Andrew X true true false false
+Xiangpeng Xiangpeng false false true true
+Raphael R false false true true
+NULL R NULL NULL NULL NULL
 
-statement ok
-drop table test;
+
+########
+## Coercion Rules
+########
 
 
-# test StringViewArray with Utf8 and Utf8View columns
 statement ok
-create table test as values ('Andrew', arrow_cast('X', 'Utf8View')),
-                            ('Xiangpeng', arrow_cast('Xiangpeng', 'Utf8View')),
-                            ('Raphael', arrow_cast('R', 'Utf8View')),
-                            (NULL, arrow_cast('R', 'Utf8View'));
+set datafusion.explain.logical_plan_only = true;
 
-query T?
-select * from test where column1 = column2;
-----
-Xiangpeng Xiangpeng
 
-query T?
-select * from test where column1 <> column2;
+# Filter should have a StringView literal and no column cast
+query TT
+explain SELECT column1_utf8 from test where column1_utf8view = 'Andrew';
 ----
-Andrew X
-Raphael R
+logical_plan
+01)Projection: test.column1_utf8
+02)--Filter: test.column1_utf8view = Utf8View("Andrew")
+03)----TableScan: test projection=[column1_utf8, column1_utf8view]
+
+# reverse order should be the same
+query TT
+explain SELECT column1_utf8 from test where 'Andrew' = column1_utf8view;
+----
+logical_plan
+01)Projection: test.column1_utf8
+02)--Filter: test.column1_utf8view = Utf8View("Andrew")
+03)----TableScan: test projection=[column1_utf8, column1_utf8view]
 
-query T?
-select * from test where column1 = arrow_cast('Andrew', 'Utf8View');
+query TT
+explain SELECT column1_utf8 from test where column1_utf8 = arrow_cast('Andrew', 'Utf8View');
```

Comment:
This plan is non ideal as it casts the column rather than the constant

Instead of 
```
CAST(test.column1_utf8 AS Utf8View) = Utf8View("Andrew")
```

the filter should be

```
test.column1_utf8 =  Utf8("Andrew")
```

I will file a ticket about this

Reply:
https://github.com/apache/datafusion/issues/10998